src/services/fileStorage/utils/filePermissionHelper.js

```diff
@@ -104,11 +58,7 @@ const checkPermissions = (permission) => async (user, file) => {
 		return Promise.reject();
 	}
 
-	return checkTeamPermission({
```

Comment:
That you removed this check altogether, does it mean, it was completely unnecessary? 

This old code is really hard to read, so I'm confused, sorry...

Reply:
That's ok 😊 . Yes there was a pretty weird (in my opinion) [permission check](https://github.com/hpi-schul-cloud/schulcloud-server/pull/4716/files#diff-0b9cbf8e7d07d5eae588c45f9b4e210d6ab2559dc9e512aa64ceae974ebef365L39) which required the author of a file to be marked as a file's creator or present in the first position of the _permissions_ array. And none of them can be true since we've had a requirement to completely get rid of all the user's data, including any references etc. And because of that we're removing any existing creator references and any personal user's reference occurrences within all the file's _permissions_ arrays. That's why we've had to also delete the requirement from the discussed function - keeping it caused some files to not being accessible anymore (by anyone basically).

Reply:
All this code is pretty pretty weird, indeed! I tried to understand it again more thoroughly but I can't get my head around it. What irritates me in your change is just, that there was a function before that checked for a team permission (at least by its name) and that this check is now completely gone. If the name is correct, this would mean that now a user from another team could see a file, that he is not allowed too, wouldn't it?

Reply:
@dyedwiper Hm when I was developing this change and testing it afterwards it felt like the whole function is not doing anything without this "creator check". But let me verify it step-by-step again and get back to you.

Reply:
@dyedwiper You were right. I was able to get rid only of this weird "author check", but keep the whole "team* role" check. Please re-review.

Reply:
Seems better, but I still don't understand what this change means. I guess somebody might not be able to see a file like he used to. But I'll approve now from a code perspective. You have to test this or wait for complaints ;)